projects/control-service/projects/job-builder/build_image.sh

```diff
@@ -6,8 +6,12 @@
 set -ex
 
 # TODO: replace those as env variables
+# Turn off debug mode temporarily to prevent credentials from being logged
+if [[ $- =~ x ]]; then debug=1; set +x; fi
 aws_access_key_id=$1
 aws_secret_access_key=$2
+[[ $debug == 1 ]] && set -x
+
 aws_region=$3
 docker_registry=$4
 git_username=$5
```

Comment:
What about git creds or registry creds? 

Would it not be simpler to disable debug mode of the shell and add prints to log what is needed.

Reply:
The focus of the story was the removal of AWS credentials. However, yes, turning off debug mode would be much better. The question is, what things then need to be logged?

Reply:
Use your own judgment. Very least the nonsensitive arguments make sense to be logged.